### PR TITLE
oop-programs_with_interesting_classes-easy-q6

### PR DESCRIPTION
**Pull request by KaiKhademi**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”

### CODE REVIEW

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
Please see my feedback on other questions in this category and apply it to this PR.

File: src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# oop-programs_with_interesting_classes-easy-q6
+
+- Parsons Question
+
+## Question Text
+
+University courses can get expensive because they often come with supply costs for things like textbooks and lab
+equipment. Make a class called `Course`.
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md:8]:
```suggestion
University courses are expensive. Beyond tuition, there are also necessary supply costs, textbooks, and lab equipment. 
Make a class called `Course` to keep track of where the costs are coming from using the requirements below.
Then make a class called `SemesterCost` to test the `Course` class.
```

File: src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    - A `double` attribute called **suppliesFee**
+    - A `double` attribute called **labFee**
+
+- Constructors:
+    - A constructor visible from another class that accepts three `double` as parameters, and uses the values
+      to initialize the **textBookFee**, **suppliesFee**, and **labFee** attributes in that order.
+    - Note that this constructor should be visible from all other classes.
+
+- Methods:
+    - A `calcTotalCost` method which returns the sum of the three attributes.
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md:26]: make a note of the visibility of this method

File: src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+### TestSocialMedia class
+
+- Main method:
+    - In the main method initalize 5 course classes and the calculate the total cost for each course using the
+      `calcTotalCost` and sum it for all of the courses and the print the total cost of all the courses.
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md:32]:
```suggestion
    - In the main method initalize 5 `Course` objects and then calculate the total cost for each course using the
      `calcTotalCost` method. Sum the total cost for all the courses and then print the semester's cost.
```

File: src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+- Constructors:
+    - A constructor visible from another class that accepts three `double` as parameters, and uses the values
+      to initialize the **textBookFee**, **suppliesFee**, and **labFee** attributes in that order.
+    - Note that this constructor should be visible from all other classes.
+
+- Methods:
+    - A `calcTotalCost` method which returns the sum of the three attributes.
+
+### TestSocialMedia class
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md:28]:
```suggestion
### SemesterCost Class
```

File: src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+University courses can get expensive because they often come with supply costs for things like textbooks and lab
+equipment. Make a class called `Course`.
+
+### Course Class
+
+The `Course` class has the following requirements:
+
+- Attributes
+    - Note that none of the attributes in this class should be visible from another class
+    - A `double` attribute called **textBookFee**
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md:16]:
```suggestion
    - A `double` attribute called **textbookFee**
```

File: src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    - A `double` attribute called **suppliesFee**
+    - A `double` attribute called **labFee**
+
+- Constructors:
+    - A constructor visible from another class that accepts three `double` as parameters, and uses the values
+      to initialize the **textBookFee**, **suppliesFee**, and **labFee** attributes in that order.
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q6/QuestionText.md:22]:
```suggestion
      to initialize the **textbookFee**, **suppliesFee**, and **labFee** attributes in that order.
```

File: src/oop/programs_with_interesting_classes/easy/q6/Course.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package oop.programs_with_interesting_classes.easy.q6;
+
+public class Course {
+    private double textBookFee;
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q6/Course.java:4]:
```suggestion
    private double textbookFee;
```

File: src/oop/programs_with_interesting_classes/easy/q6/MainTest.java [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.stream.Stream;
+
+public class MainTest {
+    // Parsons
+
+    private final String className = "Course";
+    private final String attributeName1 = "textBookFee";
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q6/MainTest.java:23]:
```suggestion
    private final String attributeName1 = "textbookFee";
```

File: src/oop/programs_with_interesting_classes/easy/q6/MainTest.java [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        Object checkupInstance = testClass.createInstance(constructorArgs);
+        _assertEquals(value1, testClass.getFieldValue(checkupInstance, attributeName1), String.format(wrongValueMessage, className, attributeName1));
+        _assertEquals(value2, testClass.getFieldValue(checkupInstance, attributeName2), String.format(wrongValueMessage, className, attributeName2));
+        _assertEquals(value3, testClass.getFieldValue(checkupInstance, attributeName3), String.format(wrongValueMessage, className, attributeName3));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependent1")
+    public void discordClassHasCorrectCalcTotalCostMethod(double value1, double value2, double value3) throws Throwable {
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q6/MainTest.java:97]:
```suggestion
    public void courseClassHasCorrectCalcTotalCostMethod(double value1, double value2, double value3) throws Throwable {
```

File: src/oop/programs_with_interesting_classes/easy/q6/MainTest.java [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+        Object output = testClass.callMethod(methodName, classInstance);
+        _assertEquals(value1 + value2 + value3, output,
+                String.format(incorrectNumDays, methodName, className));
+    }
+
+    @Test
+    @Tag("dependency1")
+    public void correctMainMethod() throws Throwable {
+        Clause[] clauses = {
+                new StringLiteral("The total cost for this semester is: 737.71 dollars"),
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q6/MainTest.java:120]: make this a variable at the top of the tests

File: src/oop/programs_with_interesting_classes/easy/q6/MainTest.java [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+                new NewLine()
+        };
+        Object classInstance = classObject.createInstance();
+        classObject.callMethod("main", new Object[][]{{new String[0], String[].class}}, new String[]{"public"}, classInstance,
+                clauses);
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q6/MainTest.java:125]: add an error message

**Review by SamamthaH — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
See Sam's review

**Pull request merged**